### ISSUE TITLE
RSA.verify null is not an object

### ISSUE THREAD

**Issue by cretumarius** [amitaymolko/react-native-rsa-native#83]
Possible Unhandled Promise Rejection (id: 0):
TypeError: null is not an object (evaluating '_reactNativeRsaNative.RSA.verify')

I'm trying to verify with publicKey if a signature signed with privateKey is trusted.
`RSA.verify(signature, payload, publicKey).then( ... )`
This should be correct regarding documentation.

Thanks in advance.

**Comment by akiladevinda** [comment]
@crafterm any solution ?

**Comment by aalhitennf** [comment]
same problem with RSA.decode

**Comment by aalhitennf** [comment]
i ditched expo and recreated my project with react native cli. problem gone.

**Comment by johhansantana** [comment]
Not sure but I think the README documentation and the actual library are different:

```typescript
export function verify(data: string, secretToVerify: string, key: string): Promise<boolean>;
```

vs what is in the README

```typescript
static verify(signature : string, message : string, keyTag : string) : Promise<boolean>
```

Given this and my little knowledge on how to verify signatures, I am confused on how to make it work like it says in the documentation since the actual library differs with the params that you're suppose to send.

**Comment by dalloglio** [comment]
> itched expo and recreated my project with react native cli. problem gone.

Yes, for me too. The problem I find is with the random byte generation in expo. The vast majority of libraries do not support applications using Expo.

**Comment by dalloglio** [comment]
For my specific problem, the solution was this library: [pagarme-card-hash](https://www.npmjs.com/package/pagarme-card-hash). No more problems with RSA.
